include/seastar/core/metrics_api.hh

```diff
@@ -396,5 +412,50 @@ struct options : public program_options::option_group {
  */
 future<> configure(const options& opts);
 
+/*!
+ * \brief Perform relabeling and operation on metrics dynamically.
+ *
+ * The general logic follows Prometheus metrics_relabel_config configuration.
+ * The relabel rules are applied one after the other.
+ * You can add or change a label. you can enable or disable a metric,
+ * in that case the metrics will not be reported at all.
+ * You can turn on and off the skip_when_empty flag.
+ *
+ * Using the Prometheus convention, the metric name is __name__.
+ * Names cannot be changed.
+ *
+ * Import notes:
+ * - The relabeling always starts from the original set of labels the metric
+ *   was created with.
+ * - calling with an empty set will remove the relabel config and will
+ *   return all metrics to their original labels
+ * - To prevent a situation that calling this function would crash the system.
+ *   in a situation where a conflicting metrics name are entered, an additional label
+ *   will be added to the labels with a unique ID.
+ *
+ * A few examples:
+ * To add a level label with a value 1, to the reactor_utilization metric:
+ *  std::vector<sm::relabel_config> rl(1);
+    rl[0].source_labels = {"__name__"};
+    rl[0].target_label = "level";
+    rl[0].replacement = "1";
+    rl[0].expr = "reactor_utilization";
+   set_relabel_configs(rl);
```

Comment:
It's nicer to use designated initializers, even in the example.

Reply:
I'm not sure what you mean, I assume that this is the common way users of the API would use it.
get a list from a file or the RestAPI, create an array and pass it.

What do you mean with designated initializers?